src/SendGrid/SendGridClient.cs

```diff
@@ -1,53 +1,16 @@
-﻿// <copyright file="SendGridClient.cs" company="Twilio SendGrid">
-// Copyright (c) Twilio SendGrid. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-// </copyright>
-
-using Newtonsoft.Json;
-using SendGrid.Helpers.Mail;
-using SendGrid.Helpers.Reliability;
-using System;
-using System.Collections.Generic;
-using System.IO;
+﻿using System.Collections.Generic;
 using System.Net;
 using System.Net.Http;
-using System.Net.Http.Headers;
-using System.Reflection;
-using System.Text;
-using System.Threading;
-using System.Threading.Tasks;
 
 namespace SendGrid
 {
     /// <summary>
-    /// A HTTP client wrapper for interacting with Twilio SendGrid's API.
+    /// An HTTP client wrapper for interacting with Twilio SendGrid's API.
     /// </summary>
-    public class SendGridClient : ISendGridClient
+    public class SendGridClient : BaseInterface
```

Comment:
I'm concerned this name (`BaseInterface`) might confuse with [C# Interfaces](https://en.wikibooks.org/wiki/C_Sharp_Programming/Interfaces). How about using `BaseClient`, similar to what you did for `BaseClientOptions`?

Reply:
That's fine with me.